src/Components/Overlay/index.jsx

```diff
@@ -2,58 +2,169 @@ import React, { useEffect, useState } from "react";
 import "./Overlay.scss";
 import { DICTIONARY_URL } from "../../constants";
 import { fetchData } from "../../util";
+import { serializeWordDefinition } from "../../serializers";
+import { notification } from "antd";
+import { Modal } from "antd";
+import { Collapse } from "antd";
+import { Typography } from "antd";
+import { Tag } from "antd";
+import { Spin } from "antd";
 
-const Overlay = ({ word, setIsOverlayVisible }) => {
-  const [definitions, setDefinition] = useState([]);
-  const [error, setError] = useState({ isSuccess: true, errorMessage: "" });
+const { Text } = Typography;
+const { Panel } = Collapse;
+
+const Overlay = ({ word, setIsModalVisible, isModalVisible }) => {
+  const [wordDetails, setWordDetails] = useState({});
+  const [partOfSpeechArray, setPartOfSpeechArray] = useState([]);
 
   useEffect(() => {
     const getWordDefinition = async () => {
       const url = DICTIONARY_URL + word;
```

Comment:
Let's move all the functions that do network requests into a new file `requests.js`.